app/selectors/post_list.js

```diff
@@ -67,7 +70,14 @@ export function makePreparePostIdsForPostList() {
 
                 // Only add the new messages line if a lastViewedAt time is set
                 const postIsUnread = post.create_at > lastViewedAt && post.user_id !== currentUser.id;
-                if (lastViewedAt !== null && !addedNewMessagesIndicator && postIsUnread && indicateNewMessages) {
+
+                if (
+                    !isSystemMessage(post) &&
```

Comment:
Is this part of the check still needed? I don't see why a post being a system message would prevent the new messages indicator being added

Reply:
Oh right, I forgot that there are also other system messages.

I've updated this PR setting post as unread when lastViewedAt is `0` (which usually means new user of a channel) and the post type is `system_combined_user_activity`.